README.md

```diff
@@ -1,2 +1,113 @@
-# front-end-code-challenge
-In this repository you can find the front-end code challenge exercise
+# Front-end-code-challenge
+
+![legend](assets/images/legend.png)
+
+
+## Objective
+To complete this code challenge, you will need to develop a simple React-based web application, containing the Legend component. In the real world scenario, this component would be a part of a more extensive flow on the map, but for the recruitment purpose, you will focus only this bit.
+
+#### DATA
+The sample legend data, located in the `data.json` file, describe the items of the legend component. It should be fetched dynamically (like would you do if it was data from a REST API).
+
+Each legend item has:
+- `id`: a unique identifier
+- `name`: the name of the layer
+- `type`: a type of the item (basic, choropleth or gradient)
+- `items`: an array of objects having name and color values
+- `description`: data to be displayed in a modal or something similar.
```

Comment:
What do you mean by "something similar"? Shouldn't we expect only the modal in this place?

Reply:
yes, you are right. Removing it